src/optimizer.js

```diff
@@ -96,12 +96,16 @@ const keywords = new Set([
  * service if any of the names are keywords or keyword-like. This is
  * safe and only prevent performance optimization.
  */
-export function getOptimizableGlobals(safeGlobal) {
-  const descs = getOwnPropertyDescriptors(safeGlobal);
-
+export function getOptimizableGlobals(globalObject, endowments = {}) {
```

Comment:
Maybe `endowmentProps` rather than `endowments`?

Reply:
Or `endowmentDescs`?

Reply:
I think it should be `endowents` since we pass `rewriterState.endowments` to it. But I'm in favor of a different name: we have `globalObject` and we have something that sits in front and hides it. I don't want to use "shadow" since people tend to associate it with proxy shadow.